includes/sys_menu.php

```diff
@@ -24,7 +25,9 @@ function header_render_hints()
 
         // Important hints:
         $hints_renderer->addHint(render_user_freeloader_hint(), true);
-        $hints_renderer->addHint(render_user_arrived_hint(), true);
+        if ($user->personalData->planned_arrival_date && (Carbon::now() > $user->personalData->planned_arrival_date)) {
+            $hints_renderer->addHint(render_user_arrived_hint(), true);
+        };
```

Comment:
The logic if a hint has to be displayed has to be in the render_user_arrived_hint(). These methods decide if the hint has to be rendered.